runtime/cudaq/qis/qvector.h

```diff
@@ -84,12 +85,20 @@ class qvector {
   //===--------------------------------------------------------------------===//
   // qvector with an initial state
   //===--------------------------------------------------------------------===//
-
-  explicit qvector(const state *);
-  explicit qvector(const state &);
-  explicit qvector(state *);
-  explicit qvector(state &);
-  explicit qvector(state &&);
+  /// @brief Construct a `qvector` from a pre-existing `state`.
+  /// This `state` could be constructed with `state::from_data` or retrieved
+  /// from an cudaq::get_state.
+  explicit qvector(const state &state) : qudits(state.get_num_qubits()) {
+    std::vector<QuditInfo> targets;
+    for (auto &q : qudits)
+      targets.emplace_back(QuditInfo{Levels, q.id()});
+    // Note: the internal state data will be cloned by the simulator backend.
+    getExecutionManager()->initializeState(targets, state.internal.get());
+  }
```

Comment:
I suppose this might be defined in the .cpp file and just declared here. Then you would only have to include the state.h file from the .cpp file...

Reply:
I'd like to do it as well. Unfortunately, `qvector` is a class template so we need to define the implementation in the header file.